DTAConfig/CustomSettings/CustomSettingFileCheckBox.cs

```diff
@@ -0,0 +1,161 @@
+using ClientCore;
+using ClientGUI;
+using Rampastring.Tools;
+using Rampastring.XNAUI;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace DTAConfig.CustomSettings
+{
+    /// <summary>
+    /// An implementation of a check-box that switches between two sets of files.
+    /// </summary>
+    public class CustomSettingFileCheckBox : XNAClientCheckBox, ICustomSetting
+    {
+        public CustomSettingFileCheckBox(WindowManager windowManager) : base(windowManager) { }
+
+        private List<FileSourceDestinationInfo> enabledFiles = new List<FileSourceDestinationInfo>();
+        private List<FileSourceDestinationInfo> disabledFiles = new List<FileSourceDestinationInfo>();
+
+        private bool EnabledFilesComplete => enabledFiles.All(f => File.Exists(f.SourcePath));
+        private bool DisabledFilesComplete => disabledFiles.All(f => File.Exists(f.SourcePath));
+
+        private bool defaultValue;
+        private bool originalState;
+        private bool restartRequired;
+        private bool checkFilePresence;
+        private bool resetUnselectableItem;
```

Comment:
Some comments here on what `checkFilePresence` and `resetUnselectableItem` precisely do (or are meant to do) would be useful.